packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -202,3 +203,28 @@ export const typeMatchesSomeSpecifier = (
   program: ts.Program,
 ): boolean =>
   specifiers.some(specifier => typeMatchesSpecifier(type, specifier, program));
+
+export function valueMatchesSpecifier(
+  node: TSESTree.Node,
+  specifier: TypeOrValueSpecifier,
+): boolean {
+  if ('name' in node && typeof node.name === 'string') {
```

Comment:
[Refactor] We don't generally like to use `in` checks for AST node narrowing. https://github.com/eslint-community/eslint-plugin-eslint-plugin/blob/main/docs/rules/no-property-in-node.md

I see from `no-deprecated` that this is only ever called with a node that is one of:

* `TSESTree.Identifier`
* `TSESTree.JSXIdentifier`
* `TSESTree.Super`

`TSESTree.Super` is always `false`. The other Identifiers have `name: string`. So you should be able to refactor this to restrict the type of `node` and not have to check` typeof node.name`. Or something along those lines. 😄

Reply:
Thanks! I fixed in [a4463a4](https://github.com/typescript-eslint/typescript-eslint/pull/10670/commits/a4463a4dae6c2c5c619d8ee63db0ac858c75ef36).